packages/wonder-blocks-dropdown/src/components/combobox.tsx

```diff
@@ -400,6 +410,33 @@ export default function Combobox({
         [selected, setSelected],
     );
 
+    React.useEffect(() => {
+        // Focus on the combobox input when the dropdown is opened.
+        if (openState) {
+            comboboxRef.current?.focus();
+        }
+    }, [openState]);
+
+    // The labels of the selected value(s).
+    const selectedLabels = React.useMemo(() => {
+        // For multiple selection, convert the selected value(s) to an array of
+        // labels.
+        if (Array.isArray(selected)) {
+            return selected.map((value) => {
+                // NOTE: Using the children prop to get the labels of the
+                // selected values, even when the list of options is filtered.
+                const item = children.find(
+                    (item) => item.props.value === value,
+                );
+                return item ? getLabel(item?.props) : "";
+            });
+        }
+
+        // Single selection mode still wraps the selected value in an array
+        // to allow for a consistent API in ComboboxLiveRegion.
+        return [labelFromSelected];
+    }, [children, labelFromSelected, selected]);
+
```

Comment:
**note:** This fixes the ordering issue with the multi-select labels. The problem was that I was iterating over the `children` array and mapping that to the labels.

```
return children
            .filter((item) => selected?.includes(item.props.value))
            .map((item) => getLabel(item.props) as string);
```
Instead of taking this approach, now I iterate over the array of `selected` values and map that to the relevant child that matches the value.

See working example here: https://5e1bf4b385e3fb0020b7073c-uwiczkggys.chromatic.com/?path=/story/packages-dropdown-combobox--multiple-selection